velox/functions/prestosql/tests/MapEntriesTest.cpp

```diff
@@ -117,3 +119,33 @@ TEST_F(MapEntriesTest, constant) {
       })));
   test::assertEqualVectors(expected, result);
 }
+
+TEST_F(MapEntriesTest, outputSizeIsBoundBySelectedRows) {
+  auto function =
+      exec::getVectorFunction("map_entries", {MAP(BIGINT(), BIGINT())}, {});
+
+  auto map = makeMapVector<int64_t, int64_t>(
+      100,
+      [](vector_size_t row) { return row % 5; },
+      [](vector_size_t row) { return row % 7; },
+      [](vector_size_t row) { return row % 11; });
+  std::vector<VectorPtr> inputsVector = {map};
+
+  // Only the first 5 rows selected.
+  SelectivityVector rows(5);
+  // This is larger than input map size but, the new rows are not selected.
+  rows.resize(1000, false);
+
+  ASSERT_EQ(rows.size(), 1000);
+  ASSERT_EQ(rows.end(), 5);
+  ASSERT_EQ(map->size(), 100);
+
+  exec::ExprSet exprSet({}, &execCtx_);
+  auto inputs = makeRowVector({});
+  exec::EvalCtx evalCtx(&execCtx_, &exprSet, inputs.get());
+  VectorPtr result;
+
+  function->apply(
+      rows, inputsVector, ARRAY(ROW({BIGINT(), BIGINT()})), evalCtx, result);
+  ASSERT_EQ(result->size(), 5);
```

Comment:
This test case may not be easy to understand without context of the bug. Could you add a comment explaining what this test ensures and what is expected?

Reply:
done